File: algorithms/number-theory/restaurant/restaurant.py
__author__ = 'jared'

from fractions import gcd
from sys import stdin, stdout

"""
Problem:
Martha is interviewing at Subway. One of the rounds of the interview
requires her to cut a bread of size l x b into smaller identical pieces such that
each piece is a square having maximum possible side length with no left over piece of bread.

Link: https://www.hackerrank.com/challenges/restaurant
"""

line = stdin.readline()
rectangle_count = int(line.strip())
for i in range(rectangle_count):
    width, height = stdin.readline().split()
    width, height = int(width), int(height)
    factor = gcd(width, height)
    squares = width / factor * height / factor
    stdout.write(str(int(squares)))
    stdout.write("\n")

File: algorithms/number-theory/restaurant/restaurant.c
#include <stdio.h>
#include <string.h>
#include <math.h>
#include <stdlib.h>

int gcd(int a, int b)
{
  int c;
  while ( a != 0 ) {
     c = a; a = b%a;  b = c;
  }
  return b;
}

int main()
{
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */

    int num_cases, width, height, factor, squares;
    fscanf(stdin, "%d\n", &num_cases);

    for (int i = 0; i < num_cases; i++) {
        fscanf(stdin, "%d %d\n", &width, &height);
        factor = gcd(width, height);
        squares = width * height / (factor * factor);
        printf("%d\n", squares);
    }

    return 0;
}